### PR TITLE
Updating src files

### PR DESCRIPTION
**Pull request by challgren**
Supporting CakePHP 3.7.0

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
 	"require":{
 		"php": ">=5.6",
 		"cakephp/cakephp": "^3.7.0",
-		"dereuromark/cakephp-shim": "^1.7.0"
+		"dereuromark/cakephp-shim": "^1.7.0",
+		"ext-mbstring": "*"
```
[the remark this answers is not in the document]
Comment (challgren) [reply composer.json]: Understood its duplicated, however PHPStorm Inspector flags all 55 occurrences as missing 'ext-mbstring' https://www.dropbox.com/s/cdtkdlhthdrvhxt/Screenshot%202019-01-01%2007.29.23.png?dl=0 which in my opinion makes the whole use of an IDE pointless if we aren't allowing the tools to flag warnings like this.  

I'll remove it.

[the remark this answers is not in the document]
Comment (dereuromark) [reply composer.json]: True. but it should also be smart enough to see the dependency imo.

**Comment by dereuromark** [comment]
Tests seem to fail.

File: composer.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (challgren) [reply composer.json]: I'll file a issue/suggestion with PHPStorm

**Comment by challgren** [comment]
Yeah I'm not sure how to fix the 1 error and the failure. Those are a little bit more advanced issues that I'm not sure how to resolve just yet.

**Comment by dereuromark** [comment]
Best to revert those then to avoid the PR to be blocked.

**Comment by challgren** [comment]
I'm 99% sure those failed before I modified a line of code. I'm running a travis build on my fork right now https://travis-ci.org/challgren/cakephp-tools/builds/474090734

**Pull request merged**